src/utils/memcxx.hpp

```diff
@@ -0,0 +1,70 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <string>
+
+#ifndef MEMCXX
+#define MEMCXX
+
+namespace memcxx_mock {
+namespace text_search {
+struct IndexContext {
+  std::string tantivyContext;  // the actual type of tantivyContext is outside the mgcxx API
+};
+
+struct IndexConfig {
+  std::string mappings;
+};
+
+struct DocumentInput {
+  std::string data;
+};
+
+struct DocumentOutput {
+  std::string data;
+};
+
+struct SearchInput {
+  std::vector<std::string> search_fields;
+  std::string search_query;
+  std::vector<std::string> return_fields;
+  std::string aggregation_query;
+};
+
+struct SearchOutput {
+  std::vector<DocumentOutput> docs;
+};
+
+// NOTE: The function names don't follow the style guide in order to be uniform with the mgcxx API
+class Mock {
```

Comment:
Move closer to the src -> `src/storage/v2/`
+ rename to `mock_mgcxx.hpp`

Reply:
Done